api/plugins/ginfo/ginfo.py

```diff
@@ -0,0 +1,153 @@
+from ginfocharacter import GinfoCharacter
+from datetime import datetime
+from firebase_pushid import PushID
+
+
+class GinfoPlugin(object):
+    """
+      GinfoPlugin provides methods to post a position to the ginfo firebase database
+    """
+    API_URL = 'https://ginfo-34baa.firebaseio.com'
+    GAME = 'conanexiles'
+    MAP = 'conanexiles_desert'
+    GINFO_USER_UID = 'y67Efa3kddSuhb26UoUapIJakMh1'
+
+    def __init__(self):
+        self.uid_generator = PushID()
+
+    def marker_path(self, group, marker):
+        """
+            Returns the path to which a marker has to be posted in the firebase database
+
+            @param group: UID of the ginfo group for which the marker should be posted
+            @param marker: UID of the marker that should be posted
+        """
+        return "games/" + self.GAME + "/groups/" + group + "/maps/" + self.MAP + '/markers/' + marker
+
+    def access_token_path(self, group, marker):
+        """
+            Returns the path to which the groups access token has to be posted in the firebase database
+
+            @param group: UID of the ginfo group for which the marker should be posted
+            @param marker: UID of the marker that should be posted
+        """
+        return "groupAccessTokenVerifier/" + group + "/" + marker
+
+    TRANSFORMATION_KX = 81.48592277749488
+    TRANSFORMATION_KY = -81.9610812970396
+    TRANSFORMATION_DX = 6275917.802625263
+    TRANSFORMATION_DY = 11160459.85122114
+
+    def linear_transform(self, x, y):
+        """
+            Applies a linear transform to the point x/y
+        """
+        return (
+            x * self.TRANSFORMATION_KX + self.TRANSFORMATION_DX,
+            y * self.TRANSFORMATION_KY + self.TRANSFORMATION_DY
+        )
+
+    def convert_position(self, x, y):
+        """
+          Converts the given position from x/y to lat/lng coordinates
+          @param x, y: x/y Position of in the game
+          @return: lat/lng for the same position on the ginfo map
+        """
+        (x, y) = self.linear_transform(x, y)
+        return SphericalMercator.unproject(x, y)
+
+    DEFAULT_MARKER_COLOR = "#1D8BF1"
+    DEFAULT_MARKER_TYPE = 1
+    DEFAULT_MARKER_SKIN = 0
+
+    def update_position(self, character, group, access_token):
+        """
+          Posts the position of this character to Firebase
+          @param character:
+                    The character for which the position should be updated
+          @param group:
+                    The group in which the marker should be postet
+          @param access_token:
+                    Access token for the ginfo group
+        """
+        (lat, lng) = self.convert_position(
+            float(character.x), float(character.y))
+
+        ginfo_character, created = GinfoCharacter.objects.get_or_create(
+            character_id=character.id)
+        if created:
+            ginfo_character.ginfo_marker_uid = self.uid_generator.next_id()
+            ginfo_character.save()
```

Comment:
This approach as discussed is more of an eventually consistency model and I'm okay with the way the code is written. Revoking my comment.